ts/features/idpay/payment/screens/IDPayPaymentResultScreen.tsx

```diff
@@ -1,34 +1,76 @@
 import { useSelector } from "@xstate/react";
 import { default as React } from "react";
-import { SafeAreaView, View } from "react-native";
-import { ContentWrapper } from "../../../../components/core/ContentWrapper";
+import { SafeAreaView, StyleSheet, View } from "react-native";
+import {
+  IOPictograms,
+  Pictogram
+} from "../../../../components/core/pictograms";
+import { VSpacer } from "../../../../components/core/spacer/Spacer";
 import { Body } from "../../../../components/core/typography/Body";
+import { NewH4 } from "../../../../components/core/typography/NewH4";
 import { IOStyles } from "../../../../components/core/variables/IOStyles";
 import FooterWithButtons from "../../../../components/ui/FooterWithButtons";
+import I18n from "../../../../i18n";
+import { PaymentFailure, PaymentFailureEnum } from "../xstate/failure";
 import { usePaymentMachineService } from "../xstate/provider";
-import { selectIsFailure, selectTransactionData } from "../xstate/selectors";
+import { selectFailureType } from "../xstate/selectors";
 
 const IDPayPaymentResultScreen = () => {
   const machine = usePaymentMachineService();
-  const transactionData = useSelector(machine, selectTransactionData);
-  const isFailure = useSelector(machine, selectIsFailure);
+  const possibleFailureType = useSelector(machine, selectFailureType);
 
   const handleClose = () => {
     machine.send("EXIT");
   };
 
-  // TODO Debug. Screen content will be added in another PR
-  const content = (
-    <Body>
-      {isFailure ? "Fallita! 🙁" : JSON.stringify(transactionData, null, 4)}
-    </Body>
+  type FailureContent = {
+    title: string;
+    subtitle?: string;
+    icon: IOPictograms;
+  };
+
+  type PaymentResults = PaymentFailure | "SUCCESS";
+  const results: Record<PaymentResults, FailureContent> = {
+    [PaymentFailureEnum.CANCELLED]: {
+      title: I18n.t("idpay.payment.result.failure.CANCELLED.title"),
+      icon: "unrecognized"
+    },
+    [PaymentFailureEnum.GENERIC]: {
+      title: I18n.t("idpay.payment.result.failure.GENERIC.title"),
+      subtitle: I18n.t("idpay.payment.result.failure.GENERIC.body"),
+      icon: "umbrella"
+    },
+    ["SUCCESS"]: {
+      title: I18n.t("idpay.payment.result.success.title"),
+      subtitle: I18n.t("idpay.payment.result.success.body"),
+      icon: "completed"
+    }
+  };
+
+  const renderContent = (resultType: PaymentResults) => (
+    <View style={resultScreenStyles.resultScreenContainer}>
+      <Pictogram name={results[resultType].icon} size={120} />
```

Comment:
```suggestion
      <Pictogram name={results[resultType].icon} size={80} />
```

Reply:
after a short discussion on Figma, 120 was approved as the uniform sizing for pictograms in this flow 👍